### ISSUE TITLE
False import-error when running as module on editable install of implicit namespace package

### ISSUE THREAD

**Issue by troiganto** [PyCQA/pylint#4251]
### Steps to reproduce

Given this project structure:
```
.
├── pyproject.toml
├── setup.cfg
├── setup.py
└── some_ns
    └── some_pkg
        ├── __init__.py
        └── some_mod.py
```

pyproject.toml
```toml
[build-system]
requires = ['setuptools', 'wheel']
build-backend = 'setuptools.build_meta'
[tool.pylint]
```

setup.cfg
```cfg
[metadata]
name = minimization
version = 1.0
[options]
packages = find_namespace:
install_requires = pylint==2.7.3.dev1
```

setup.py
```python
"""setup.py for editable installs"""
import setuptools
if __name__ == "__main__":
    setuptools.setup()
```

some_ns/some_pkg/__init__.py
```python
"""Some package."""
from .some_mod import SomeClass
```

some_ns/some_pkg/some_mod.py
```python
"""Some module."""


class SomeClass(Exception):
    """Some class."""
```

Set up a virtual environment as follows:
```bash
python3 -m venv venv
source venv/bin/activate
pip install git+https://github.com/PyCQA/pylint.git@2.7
pip install --editable .
```

### Current behavior

Result of `python -m pylint some_ns/`:
```
************* Module some_ns.some_pkg
some_ns/some_pkg/__init__.py:2:0: E0401: Unable to import 'some_ns.some_pkg.some_mod' (import-error)

----------------------------------------------------------------------
Your code has been rated at -15.00/10 (previous run: -15.00/10, +0.00)
```

Result of `pylint some_ns/`:
```
--------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 10.00/10, +0.00)
```

### Expected behavior

Expected both invocations to not find any import-errors.

### Result of `pylint --version` output:
```
pylint 2.7.3-dev1
astroid 2.5.1
Python 3.8.6 (default, Jan 27 2021, 15:42:20) 
[GCC 10.2.0]
```
(Same for `python -m pylint --version`)

### Additional notes

- error does not appear with regular install
- error does not appear with non-namespace package
- error does not appear on PyLint 2.7.1 or earlier
- I _think_ but cannot prove that this is related to #4153 

Output of `pylint --init-hook='print(sys.path)' some_ns`:
```python
['PROJECT_DIR/venv/bin', '/usr/lib/python38.zip', '/usr/lib/python3.8', '/usr/lib/python3.8/lib-dynload', 'PROJECT_DIR/venv/lib/python3.8/site-packages', 'PROJECT_DIR', 'PROJECT_DIR/venv/lib/python3.8/site-packages/astroid/brain']
```

Output of `python -m pylint --init-hook='print(sys.path)' some_ns`:
```python
['/usr/lib/python38.zip', '/usr/lib/python3.8', '/usr/lib/python3.8/lib-dynload', 'PROJECT_DIR/venv/lib/python3.8/site-packages', 'PROJECT_DIR/venv/lib/python3.8/site-packages/astroid/brain']
```
This one is missing `PROJECT_DIR/venv/bin` and `PROJECT_DIR` entries

**Comment by troiganto** [comment]
Ah! I just tried `pylint-2.8.0.dev1` (installed from master branch at commit d274b660a7b7cc4b6910bf69a33dae0a7d5d9c53) and the error seems to not exist there. So the issue might've been solved somewhere between 2.7.3 and 2.8.

**Comment by cdce8p** [comment]
I believe this was fixed in #4164

**Comment by troiganto** [comment]
This looks right indeed! Based on the changelog in the linked PR, I assume this will be backported into the 2.7 branch before the 2.7.3 release?

**Comment by cdce8p** [comment]
@troiganto Yeah, this will be included in the next release, `2.7.3`

**Comment by troiganto** [comment]
Wonderful, gonna close this issue then. Thanks for the quick response and sorry for the noise!
